File: sd/trabalhoFinal/final/esqueleto.py
import json
from database import Database
from filme import Filme

class Esqueleto:
    def __init__(self):
        self.db = Database()

    def adicionar_filme(self, arguments_json):
        arguments = json.loads(arguments_json)
        filme = Filme(**arguments)
        return self.db.adicionarFilme(filme)

    def remover_filme(self, arguments_json):
        arguments = json.loads(arguments_json)
        return self.db.removerFilme(arguments)

    def exibir_detalhe(self, arguments_json):
        arguments = json.loads(arguments_json)
        return self.db.exibirDetalhe(arguments)

    def mostrar_catalogo(self):
        return self.db.mostrarCatalogo()

File: sd/trabalhoFinal/final/src/server/Esqueleto.java
package org.example;
import java.sql.SQLException;

public class Esqueleto {
    private Database db;

    public Esqueleto(Database db) {
        this.db = db;
    }

    public String adicionarFilme(String arguments) {
        Filme filme = Filme.fromJson(arguments);
        try {
            return db.adicionarFilme(filme);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public String removerFilme(String arguments) {
        int id = Integer.parseInt(arguments);
        try {
            return db.removerFilme(id);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public String exibirDetalhe(String arguments) {
        int id = Integer.parseInt(arguments);
        try {
            return db.exibirDetalhe(id);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public String mostrarCatalogo() {
        try {
            return db.mostrarCatalogo();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}